lua/harpoon/list.lua

```diff
@@ -156,6 +158,8 @@ function HarpoonList:resolve_displayed(displayed)
     end
 
     self.items = new_list
+
+    require("harpoon"):sync()
```

Comment:
sync should only be called when vim is about to exit.  also list should not know of harpoon (at least my intention is that this exists by itself)